src/timedisc/timedisc.jl

```diff
@@ -1,4 +1,155 @@
 
+# The following structures and methods provide a minimal implementation of
+# the low-storage explicit Runge-Kutta method of
+# Carpenter, Kennedy (1994) Fourth order 2N storage RK schemes, Solution 3
+# using the same interface as OrdinaryDiffEq.jl.
+struct SimpleAlgorithm2N45
+  a::SVector{5, Float64}
+  b::SVector{5, Float64}
+  c::SVector{5, Float64}
+
+  function SimpleAlgorithm2N45()
+    a = @SVector [0.0, 567301805773.0 / 1357537059087.0,2404267990393.0 / 2016746695238.0,
+       3550918686646.0 / 2091501179385.0, 1275806237668.0 / 842570457699.0]
+    b = @SVector [1432997174477.0 / 9575080441755.0, 5161836677717.0 / 13612068292357.0,
+        1720146321549.0 / 2090206949498.0, 3134564353537.0 / 4481467310338.0,
+        2277821191437.0 / 14882151754819.0]
+    c = @SVector [0.0, 1432997174477.0 / 9575080441755.0, 2526269341429.0 / 6820363962896.0,
+        2006345519317.0 / 3224310063776.0, 2802321613138.0 / 2924317926251.0]
+
+    new(a, b, c)
+  end
+end
+
+mutable struct SimpleIntegrator2NOpts{Callback}
+  callback::Callback
+  adaptive::Bool
+  dtmax::Float64
+  tstops::Vector{Float64}
+end
+
+function SimpleIntegrator2NOpts(callback, tspan; kwargs...)
+  SimpleIntegrator2NOpts{typeof(callback)}(
+    callback, false, Inf, [last(tspan)])
+end
+
+mutable struct SimpleIntegrator2N{RealT<:Real, uType, Params, Sol, Alg, SimpleIntegrator2NOpts}
+  u::uType
+  du::uType
+  u_tmp::uType
+  t::RealT
+  dt::RealT
+  dtcache::RealT
+  iter::Int
+  p::Params
+  sol::Sol
+  alg::Alg
+  opts::SimpleIntegrator2NOpts
+  finalstep::Bool
+end
+
+function solve(ode::ODEProblem, alg::SimpleAlgorithm2N45;
+               dt, callback=nothing, kwargs...)
+  u = similar(ode.u0)
+  du = similar(u)
+  u_tmp = similar(u)
+  t = first(ode.tspan)
+  iter = 0
+  integrator = SimpleIntegrator2N(u, du, u_tmp, t, dt, zero(dt), iter, ode.p,
+                  (prob=ode,), alg,
+                  SimpleIntegrator2NOpts(callback, ode.tspan; kwargs...), false)
+  init!(integrator)
+  solve!(integrator)
+end
+
+function init!(integrator::SimpleIntegrator2N)
```

Comment:
Needs... high-level documentation... ;-)

Also, why the need for an `init!` method, if we call it from `solve` anyways? Or, put differently, why can't we directly pass this algorithm to the "normal" solve? Or... why does have the solve above even initialization stuff when there is an `init!` later on? It probably all makes sense, I just don't get it (yet)

Reply:
I've moved the initialization into `solve` in a0ca542.

Reply:
:+1: